temporal-sdk/src/main/java/io/temporal/internal/statemachines/UpdateProtocolStateMachine.java

```diff
@@ -199,8 +201,10 @@ public void accept() {
         Acceptance.newBuilder()
             .setAcceptedRequestMessageId(requestMsgId)
             .setAcceptedRequestSequencingEventId(requestSeqID)
+            .setAcceptedRequest(originalRequest.get())
             .build();
-
+    // Clear the original request to allow GC to reclaim the memory.
+    originalRequest = Optional.empty();
```

Comment:
Should do this on reject too correct? Or is there no reason because reject removes this whole state machine?

Reply:
reject is a terminal state so the whole state machine will be cleaned up